### ISSUE TITLE
Better explanation for the "Too many chat channels" message

### ISSUE THREAD

**Issue by AfroThundr3007730** [Chatterino/chatterino2#5424]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

Whenever I open chatterino, I get the following message on some of the opened channel tabs:

```
You are connected to too many chat channels. Please close some existing chat connections to join a new chat
```

This only started happening after the last chatterino update. Before this, there was no issues connecting to as many channels as I want. I have over 100 tabs open in my current setup to keep track of all the channels I frequently visit.

What changed? Did Twitch impose a new API limit affecting this, or is this just client side? How can this be worked around? What even is the limit anyway?

### OS and Chatterino Version

Chatterino 2.5.1 (commit 3aead09) built with Qt 6.7.0 Running on Fedora Linux 40 (Forty), kernel: 6.8.9-300.fc40.x86_64

**Comment by Felanbird** [comment]
> This only started happening after the last chatterino update.

This is a Twitch restriction of 100 channels.

**Comment by AfroThundr3007730** [comment]
Well that is not optimal. I suppose I'll need to use multiple accounts then. I see chatterino can login with multiple accounts but how is this used? I don't see any option for choosing an account on the tabs I have open, nor when creating new tabs/splits.

Right now most of the channels appear to be using my alt and I still see the channel limit message on other tabs. Is chatterino only able to use one of the accounts at a time? And how is the switching accomplished?

**Comment by AfroThundr3007730** [comment]
I just thought of something else. Is there a way to make chatterino not join a channel until it's actually live? Even though I have a bunch open, it's usually only a couple dozen concurrent live channels in total. That plus maybe joining offline channels when the tab is selected should be sufficient, I'd think. Of course, that does mean offline chat messages for channels that aren't selected would be missed.

**Comment by Mm2PL** [comment]
> Well that is not optimal. I suppose I'll need to use multiple accounts then.
> Right now most of the channels appear to be using my alt and I still see the channel limit message on other tabs. Is chatterino only able to use one of the accounts at a time? And how is the switching accomplished?

We cannot suggest you bypass the rate limits. You can only be connected with one account at a time in a single instance of the program. You can change which account that is with the User button by the tabs. 

> Is there a way to make chatterino not join a channel until it's actually live?

No we currently do not support anything like that. Channels can have live notifications even if they are not open though.

**Comment by AfroThundr3007730** [comment]
> No we currently do not support anything like that. Channels can have live notifications even if they are not open though.

So if I don't have a tab for a particular channel I can still get a notification in chatterino when they're live? If so I'd like to find out how to get that configured.

**Comment by Mm2PL** [comment]
> So if I don't have a tab for a particular channel I can still get a notification in chatterino when they're live? If so I'd like to find out how to get that configured.

You can just add the channel to your live notifications. Either through the split header menu (... -> More -> Notify when live) or the settings:
![image](https://github.com/Chatterino/chatterino2/assets/25011746/d265cc47-9ea3-4a92-8c7a-4782c1af10b2)

**Comment by AfroThundr3007730** [comment]
Ah ok, I thought so.

> Either through the split header menu (... -> More -> Notify when live) 

I take it that option adds to the above list? Well that'll do fine then. I'm assuming these will show up in the /live tab then, which works for me. Now to import all these channels.

**Comment by AfroThundr3007730** [comment]
Thanks for the help, by the way.

**Comment by mjeveritt** [comment]
Fell into this trap when I scripted up joining ALL my followed channels on Twitch into Chatterino, to take advantage of Live notifications (oops!). Which, at time of writing is about 160 channels... I will duly fix up dropping some [older] channels, and adding to the Live config option, as suggested here.

I did come across [this article](https://discuss.dev.twitch.com/t/get-date-of-last-broadcast/23944) which fits nicely into this 'dilemma'.. about finding out who are the 'newest'/'oldest' streamers based on last-live date-time . Given Chatterino already tracks this data, it ought to be possible to figure this out.

Whilst I'm not going to hijack this Issue, I will propose a Feature-Request that ties in with it, to pre-select the 'most-recent' streamers to 'attach' to the max. chat channel limit (ie. 100 tabs) and keep the rest in a 'bank' that makes use of the Live Notifications feature. I'll defer to the author what the best storage method might be for this... might need a slight restructure!
